consensus/src/liveness/round_state.rs

```diff
@@ -256,13 +263,21 @@ impl RoundState {
             self.vote_sent = None;
             self.timeout_sent = None;
             let timeout = self.setup_timeout(1);
+
+            let (prev_round_timeout_votes, prev_round_timeout_reason) = prev_round_timeout_votes
+                .map(|votes| votes.unpack_aggregate(verifier))
+                .unzip();
+
             // The new round reason is QCReady in case both QC.round + 1 == new_round, otherwise
             // it's Timeout and TC.round + 1 == new_round.
             let new_round_reason = if sync_info.highest_certified_round() + 1 == new_round {
                 NewRoundReason::QCReady
             } else {
-                NewRoundReason::Timeout
+                let prev_round_timeout_reason =
+                    prev_round_timeout_reason.expect("TC must be present for timeout round.");
```

Comment:
we don't necessarily aggregate TC ourselves, the TC could be from the sync info?

Reply:
The expect message might be misleading, but we are just trying to form a reason with locally available info. If we don't have enough f+1 votes, the reason will be Timeout::Unknown.

Reply:
I still don't think this works, prev_round_timeout_votes can be None while we can still enter here if sync info carries a TC and that would directly panic.

Reply:
Ah, I see what you mean, my bad. Yeah, will fix it.